### ISSUE TITLE
can`t deploy my hexo bu hexo-deployer-git,please help me

### ISSUE THREAD

**Issue by weiyouwozuiku** [hexojs/hexo-deployer-git#186]
when i run the command :"npx hexo d -g".something don`t work.But,when i just run "hexo server ",it can run as usual.
this is the error log:
```
(node:106578) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:106578) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:106578) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
(node:106578) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(node:106578) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:106578) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
INFO  Start processing
INFO  Files loaded in 626 ms
INFO  0 files generated in 2.43 s
INFO  Deploying: git
INFO  Clearing .deploy_git folder...
INFO  Copying files from public folder...
FATAL Something's wrong. Maybe you can find the solution here: https://hexo.io/docs/troubleshooting.html
TypeError [ERR_INVALID_ARG_TYPE]: The "mode" argument must be integer. Received an instance of Object
    at copyFile (fs.js:1890:10)
    at tryCatcher (/home/king/blog/MyBlog/node_modules/bluebird/js/release/util.js:16:23)
    at ret (eval at makeNodePromisifiedEval (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promisify.js:184:12), <anonymous>:13:39)
    at /home/king/blog/MyBlog/node_modules/hexo-fs/lib/fs.js:144:39
    at tryCatcher (/home/king/blog/MyBlog/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:547:31)
    at Promise._settlePromise (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:729:18)
    at Promise._fulfill (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:673:18)
    at Promise._resolveCallback (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:466:57)
    at Promise._settlePromiseFromHandler (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:559:17)
    at Promise._settlePromise (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:729:18)
    at Promise._fulfill (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:673:18)

```

hexo version
```
hexo version                                                                                  
(node:106770) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:106770) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:106770) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
(node:106770) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(node:106770) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:106770) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
hexo: 4.2.0
hexo-cli: 3.1.0
os: Linux 4.19.121-1-MANJARO linux x64
node: 14.2.0
v8: 8.1.307.31-node.33
uv: 1.37.0
zlib: 1.2.11
brotli: 1.0.7
ares: 1.16.0
modules: 83
nghttp2: 1.40.0
napi: 6
llhttp: 2.0.4
openssl: 1.1.1g
cldr: 37.0
icu: 67.1
tz: 2019c
unicode: 13.0

```
`cat package.json`
```
{
  "name": "hexo-site",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "build": "hexo generate",
    "clean": "hexo clean",
    "deploy": "hexo deploy",
    "server": "hexo server"
  },
  "hexo": {
    "version": "4.2.0"
  },
  "dependencies": {
    "hexo": "^4.0.0",
    "hexo-asset-image": "^1.0.0",
    "hexo-deployer-git": "^2.1.0",
    "hexo-filter-github-emojis": "^2.1.0",
    "hexo-filter-mermaid-diagrams": "^1.0.5",
    "hexo-generator-archive": "^1.0.0",
    "hexo-generator-category": "^1.0.0",
    "hexo-generator-index": "^1.0.0",
    "hexo-generator-search": "^2.4.0",
    "hexo-generator-tag": "^1.0.0",
    "hexo-permalink-pinyin": "^1.1.0",
    "hexo-prism-plugin": "^2.3.0",
    "hexo-renderer-ejs": "^1.0.0",
    "hexo-renderer-marked": "^2.0.0",
    "hexo-renderer-stylus": "^1.1.0",
    "hexo-server": "^1.0.0",
    "hexo-wordcount": "^6.0.1"
  }
}

```

**Comment by stevenjoezhang** [comment]
Upgrade to Hexo v4.2.1

**Comment by weiyouwozuiku** [comment]
`cat package.json`
```
{
  "name": "hexo-site",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "build": "hexo generate",
    "clean": "hexo clean",
    "deploy": "hexo deploy",
    "server": "hexo server"
  },
  "hexo": {
    "version": "4.2.1"
  },
  "dependencies": {
    "hexo": "^4.2.1",
    "hexo-asset-image": "^1.0.0",
    "hexo-deployer-git": "^2.1.0",
    "hexo-filter-github-emojis": "^2.1.0",
    "hexo-filter-mermaid-diagrams": "^1.0.5",
    "hexo-generator-archive": "^1.0.0",
    "hexo-generator-category": "^1.0.0",
    "hexo-generator-index": "^1.0.0",
    "hexo-generator-search": "^2.4.0",
    "hexo-generator-tag": "^1.0.0",
    "hexo-permalink-pinyin": "^1.1.0",
    "hexo-prism-plugin": "^2.3.0",
    "hexo-renderer-ejs": "^1.0.0",
    "hexo-renderer-marked": "^2.0.0",
    "hexo-renderer-stylus": "^1.1.0",
    "hexo-server": "^1.0.0",
    "hexo-wordcount": "^6.0.1"
  }
}%                       
```
error message
```
(node:4963) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:4963) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:4963) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
(node:4963) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(node:4963) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:4963) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
INFO  Start processing
INFO  Files loaded in 1.02 s
INFO  0 files generated in 2.85 s
INFO  Deploying: git
INFO  Clearing .deploy_git folder...
INFO  Copying files from public folder...
FATAL Something's wrong. Maybe you can find the solution here: https://hexo.io/docs/troubleshooting.html
TypeError [ERR_INVALID_ARG_TYPE]: The "mode" argument must be integer. Received an instance of Object
    at copyFile (fs.js:1890:10)
    at tryCatcher (/home/king/blog/MyBlog/node_modules/bluebird/js/release/util.js:16:23)
    at ret (eval at makeNodePromisifiedEval (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promisify.js:184:12), <anonymous>:13:39)
    at /home/king/blog/MyBlog/node_modules/hexo-fs/lib/fs.js:144:39
    at tryCatcher (/home/king/blog/MyBlog/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:547:31)
    at Promise._settlePromise (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:729:18)
    at Promise._fulfill (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:673:18)
    at Promise._resolveCallback (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:466:57)
    at Promise._settlePromiseFromHandler (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:559:17)
    at Promise._settlePromise (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:604:18)
    at Promise._settlePromise0 (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:649:10)
    at Promise._settlePromises (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:729:18)
    at Promise._fulfill (/home/king/blog/MyBlog/node_modules/bluebird/js/release/promise.js:673:18)

```

**Comment by YoshinoriN** [comment]
refs: https://github.com/hexojs/hexo/issues/4281

**Comment by YoshinoriN** [comment]
@weiyouwozuiku 
It seems caused by [hexo-fs](https://github.com/hexojs/hexo-deployer-git/blame/c312a1507a8136ca6bba23949cbf7bba55d739d7/package.json#L41). hexo-deployer-git depends on hexo-fs and hexo-fs@2.0.0 is incompatible with Node.js 14.

We are already released `hexo-fs@2.0.1` that is support Node.js 14.
Would you please `hexo-deployer-git` once and re-install again?

I think, after re-install `hexo d` will work well.

**Comment by weiyouwozuiku** [comment]
thanks,my problom fixed.

**Comment by ousugo** [comment]
> @weiyouwozuiku
> It seems caused by [hexo-fs](https://github.com/hexojs/hexo-deployer-git/blame/c312a1507a8136ca6bba23949cbf7bba55d739d7/package.json#L41). hexo-deployer-git depends on hexo-fs and hexo-fs@2.0.0 is incompatible with Node.js 14.
> 
> We are already released `hexo-fs@2.0.1` that supports Node.js 14.
> Would you please re-install `hexo-deployer-git`?
> 
> I think, after re-install `hexo d` will work well.

same problem with Node.js14.

but re-install `hexo-deployer-git` don't work for me,  but `npm install hexo-fs` work.

**Comment by printempw** [comment]
It seems that `hexo-fs@2.0.1` does not work on Node.js 14.

The latest published version `hexo-deployer-git@2.1.0` still depends on `hexo-fs@^2.0.0`, which won't resolve to `hexo-fs@3.1.0`.

Override the dependency with [yarn `resolutions`](https://github.com/printempw/printempw.github.io/commit/f17e995cd3af66a0721b532a2420907c75cd0fac) solved the problem and now it works on Node.js 14.

Looking forward to a new release.
